extensions/http-curl/client/HTTPClient.cpp

```diff
@@ -327,7 +356,9 @@ void HTTPClient::set_request_method(const std::string method) {
   if (my_method == "POST") {
     curl_easy_setopt(http_session_, CURLOPT_POST, 1L);
   } else if (my_method == "PUT") {
-    curl_easy_setopt(http_session_, CURLOPT_PUT, 0L);
+    curl_easy_setopt(http_session_, CURLOPT_UPLOAD, 1L);
+  } else if (my_method == "HEAD") {
+    curl_easy_setopt(http_session_, CURLOPT_NOBODY, 1L);
```

Comment:
Just passing "HEAD" as the CURLOPT_CUSTOMREQUEST will not work, because curl will expect a response body. CURLOPT_NOBODY is the proper way to do it.

Reply:
I don't think HEAD was ever completed in the client -- this is not a fully complete client and I was hoping we'd switch to https://github.com/jpbarrette/curlpp, where we simply wrapped their code in our HTTPClient. As per the comment, above, do you think this is advantageous? My hope was that we'd do that for 1.0, but if there is consensus maybe we should use their client sooner, especially since cycles to fix bugs ( known and not known ) in our http client seem wasteful. Wrapping curlpp may be ultimately the best solutions.

Reply:
I understand that this is not a complete client, and that HEAD requests have not been previously used in the code. However, to test the HEAD functionality of ListenHTTP it has to be added now.
I think it would be best to discuss curlpp in the issue you created. I have not used it, and in general just adding another layer of API which we can be use either properly or badly is not a solution in itself - I will have to review the project to decide for myself whether it would have significant advantages over just using the curl API.
What I am sure I don't want to do is introduce a new, untested thirdparty dependency in 0.6.1, and as this is a very trivial fix in the client, I don't think it warrants that.